src/library/Pagination/Pagination.js

```diff
@@ -0,0 +1,248 @@
+      marginBottom: `-${theme.Pagination_gutterWidth}`,
+
+      '& > *': {
+        marginBottom: theme.Pagination_gutterWidth
+      }
+    };
+  },
+  {
+    includeStyleReset: true,
+    withProps: {
+      element: 'nav',
+      justifyContent: 'end'
+    }
+  }
+);
+const PageSizer = createStyledComponent(_PageSizer, styles.pageSizer);
+
+/**
+ * Pagination offers a means to control the space consumed by a large data set
+ * or other collection of items by limiting the page size and providing
+ * navigation for access to all pages.
+ */
+class Pagination extends PureComponent<Props> {
+  static displayName = 'Pagination';
+  static defaultProps = {
+    category: 'rows',
+    messages: {
+      pagination: { label: 'Pagination', next: 'Next', previous: 'Previous' },
+      pageJumper: {
+        label: 'Jump to page',
+        placeholder: 'Page #'
+      },
+      pageSizer: {
+        status: (category, first, last, total) =>
+          `${first}–${last} of ${total} ${category}`,
+        itemText: (pageSize) => `${pageSize} per page`
+      }
+    },
+    pageSizes: [10, 20, 50],
+    showPageNumbers: true,
+    size: 'medium',
+    visibleRange: 5
+  };
+
+  pageJumperInput: ?HTMLInputElement;
+
+  render() {
+    const {
+      category,
+      currentPage,
+      messages,
+      showPageJumper,
+      pageSize,
+      showPageSizer,
+      pageSizes,
+      size,
+      theme: baseTheme,
+      totalCount,
+      ...restProps
+    } = this.props;
+    const theme = componentTheme(baseTheme);
+
+    const rootProps = {
+      'aria-label': messages.pagination.label,
+      gutterWidth: theme.Pagination_gutterWidth,
+      ...restProps
+    };
+
+    const totalPages = this.totalPages(pageSize);
+
+    const pagesProps = {
+      handleClick: this.handleClick,
+      handleIncrement: this.handleIncrement,
+      totalPages,
+      ...this.props
+    };
+
+    const content = [<Pages key="Pages" {...pagesProps} />];
+
+    if (showPageJumper) {
+      const pageJumperProps = {
+        currentPage,
+        inputRef: this.setPageJumperRef,
+        key: 'Page Jumper',
+        messages,
+        onPageChange: this.handlePageChange,
+        size,
+        totalPages,
+        width: theme.PaginationPageJumper_width
+      };
+      content.unshift(<PageJumper {...pageJumperProps} />);
+    }
+
+    if (showPageSizer) {
+      const pageSizerProps = {
+        currentPage,
+        category,
+        key: 'Page Sizer',
+        messages,
+        onPageChange: this.handlePageChange,
+        onPageSizeChange: this.handlePageSizeChange,
+        pageSize,
+        pageSizes,
+        size,
+        totalCount,
+        totalPages
+      };
+      content.unshift(<PageSizer {...pageSizerProps} />);
+    }
+
+    return <Root {...rootProps}>{content}</Root>;
+  }
+
+  setPageJumperRef = (node: ?HTMLInputElement) => {
+    this.pageJumperInput = node;
+  };
+
+  totalPages = (pageSize) => Math.ceil(this.props.totalCount / pageSize);
```

Comment:
Can we maybe rename this to `getTotalPages`, `calcTotalPages` or something similar?